jsapp/js/components/common/koboDropdown.tsx

```diff
@@ -1,27 +1,52 @@
 import React from 'react';
 import ReactDOM from 'react-dom';
-import autoBind from 'react-autobind';
 import bem, {makeBem} from 'js/bem';
 import {
-  createEnum,
   KEY_CODES,
 } from 'js/constants';
 import koboDropdownActions from './koboDropdownActions';
 import './koboDropdown.scss';
 
-export const KOBO_DROPDOWN_THEMES = createEnum([
-  'light',
-  'dark',
-]);
+export enum KOBO_DROPDOWN_THEMES {
+  light = 'light',
+  dark = 'dark',
+}
+
+export enum KOBO_DROPDOWN_PLACEMENTS {
+  'up-left' = 'up-left',
+  'up-center' = 'up-center',
+  'up-right' = 'up-right',
+  'down-left' = 'down-left',
+  'down-center' = 'down-center',
+  'down-right' = 'down-right',
+}
 
-export const KOBO_DROPDOWN_PLACEMENTS = createEnum([
-  'up-left',
-  'up-center',
-  'up-right',
-  'down-left',
-  'down-center',
-  'down-right',
-]);
+type KoboDropdownThemesType = 'light' | 'dark'
+
+type KoboDropdownPlacementsType =
+  'up-left' |
+  'up-center' |
+  'up-right' |
+  'down-left' |
+  'down-center' |
+  'down-right'
+
+
+type KoboDropdownProps = {
+  theme: KoboDropdownThemesType,
+  placement: KoboDropdownPlacementsType,
+  isDisabled: boolean, // disables the dropdowns trigger, thus disallowing opening dropdown
+  hideOnMenuClick: boolean, // hides menu whenever user clicks inside it, useful for simple menu with a list of actions
+  hideOnMenuOutsideClick: boolean, // hides menu when user clicks outside it
+  hideOnEsc: boolean, // hides menu when opened and user uses Escape key
+  triggerContent: React.ReactNode,
+  menuContent: React.ReactNode, // the content of dropdown, anything's allowed
```

Comment:
I discovered recently that we can use JSDoc for even tiniest part of code for TS to display them. So please change all those to:

```ts
/** the content of dropdown, anything's allowed */
menuContent: React.ReactNode,
```